packages/accordion/src/useAccordion.js

```diff
@@ -6,19 +6,29 @@
  */
 
 import { useState } from 'react';
-import { composeEventHandlers, generateId, KEY_CODES } from '@zendeskgarden/container-utilities';
+import {
+  composeEventHandlers,
+  generateId,
+  KEY_CODES,
+  getControlledValue
+} from '@zendeskgarden/container-utilities';
 
 export function useAccordion({
   idPrefix,
-  expandedSections = [],
+  expandedSections,
+  onChange,
   expandable = true,
   collapsible = true
 } = {}) {
-  const [prefix] = useState(idPrefix || generateId('garden-field-container'));
+  const [prefix] = useState(idPrefix || generateId('garden-accordion-container'));
   const TRIGGER_ID = `${prefix}--trigger`;
   const PANEL_ID = `${prefix}--panel`;
-  const [expandedState, setExpandedState] = useState(expandedSections);
-  const [disabledState, setDisabledState] = useState(collapsible ? [] : expandedSections);
+  const [expandedState, setExpandedState] = useState(expandedSections || []);
+
+  const controlledExpandedState = getControlledValue(onChange && expandedSections, expandedState);
```

Comment:
`controlledExpandedState` is an inaccurate variable name because it holds an uncontrolled value when an `onChange` isn't passed into the `useAccordion` hook.

Also, an uncontrolled component is driven by _state_, while a controlled component is driven by _props_. Both uncontrolled and controlled components should probably have an `onChange` callback.

Reply:
The current implementation's uncontrolled usage only works when an `onChange` isn't passed. This can be problematic because an uncontrolled Accordion component will need an `onChange`.